connection.go

```diff
@@ -391,6 +398,12 @@ func (conn *Connection) IsClosed() bool {
 	}
 }
 
+func (conn *Connection) releasePacket(pp *BinaryPacket) {
+	if conn.poolMaxPacketSize < cap(pp.body) || conn.poolMaxPacketSize == 0 {
```

Comment:
it would be better to swap conditions.

Reply:
Agreed. And/or rewrite the function to return early if the release condition fails.

Reply:
swapped conditions. Now poolMaxPacketSize comes first.